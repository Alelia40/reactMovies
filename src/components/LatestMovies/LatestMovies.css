.movieDisplay {
    overflow: auto;
    width: 100%;
}

.movieCard {
    width: 18rem;
    margin-bottom: 10px;
}

.bookNowButton {
    width: 75%;
    margin-bottom: 5px;
}

.detailsLink {
    color: #6c757d;
}
.detailsLink:hover {
    color: #28a745;
}