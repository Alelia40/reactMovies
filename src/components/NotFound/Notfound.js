import React from 'react'

import {Container} from 'react-bootstrap';

function Notfound() {
    return (
        <Container>
            <h3>404 Not Found</h3>
            <br></br>
            <a href="/">Return Home</a>
        </Container>
    )
}

export default Notfound
