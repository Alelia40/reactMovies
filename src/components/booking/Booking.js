import React, { useEffect, useState } from 'react'
import { Alert, Card, Container, Button, Row, Col, Form, InputGroup } from 'react-bootstrap';

import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { fetchDetails } from '../../redux/actions/Index';

import { useLocation, useParams } from 'react-router-dom';

import { FiChevronLeft, FiBookOpen, FiMinus, FiPlus } from 'react-icons/fi';

import './Booking.css';

function Booking(props) {

    const { id } = useParams();
    let selectedTicket = new URLSearchParams(useLocation().search).get("seat");

    const [numTickets, setNumTickets] = useState(1);
    const [email, setEmail] = useState("");
    const [name, setName] = useState("");

    useEffect(() => {
        console.log(id + "," + selectedTicket);
        props.fetchDetails(id);
    }, [])

    return (
        <div id="bookingWrapper">
            <Container>
                {props.loading &&
                    <span>LOADING ...</span>
                }
                {props.data &&
                    <Card className="bookingCard" bg="light" text="secondary" border="secondary">
                        <Card.Header>Purchasing Tickets</Card.Header>
                        <Card.Body>
                            <Row>
                                <Col>
                                    <Card.Title>Movie: {props.data.title}</Card.Title>
                                </Col>
                                <Col>
                                    <Card.Title>Show Date: {new Date().toLocaleDateString('en-US')}</Card.Title>
                                </Col>
                            </Row>
                            <hr></hr>
                            <Card.Title>Select Options</Card.Title>
                            <hr></hr>
                            <Form>
                                <Form.Group>
                                    <Form.Label>Number Of Tickets ({selectedTicket.toUpperCase()})</Form.Label>
                                    <Row>
                                        <Col>
                                            <Button variant="outline-secondary" onClick={() => setNumTickets(numTickets - 1)} disabled={numTickets <= 1}><FiMinus /></Button>
                                            <span style={{ marginLeft: '10px', marginRight: '10px' }}>{numTickets}</span>
                                            <Button variant="outline-secondary" onClick={() => setNumTickets(numTickets + 1)} disabled={numTickets >= 10}><FiPlus /></Button>
                                        </Col>
                                    </Row>
                                </Form.Group>
                                <Form.Group>
                                    <Form.Label>Showtime</Form.Label>
                                    <Form.Control as="select" multiple>
                                        <option>7:00</option>
                                        <option>7:30</option>
                                        <option>8:00</option>
                                        <option>8:30</option>
                                        <option>9:00</option>
                                        <option>9:30</option>
                                        <option>10:00</option>
                                    </Form.Control>
                                </Form.Group>
                                <Form.Group>
                                    <Form.Label>* Email address</Form.Label>
                                    <Form.Control type="email" placeholder="name@example.com" />
                                </Form.Group>
                            </Form>
                            <hr></hr>
                            <Card.Title>Booking Cost: {eval(props.data.prices[selectedTicket] * numTickets) + " " + props.data.currency}</Card.Title>
                            <Button variant="outline-success">Book Now <FiBookOpen /></Button>
                        </Card.Body>
                        <Card.Footer>
                            <Card.Link className="detailsLink"><FiChevronLeft /> Cancel</Card.Link>
                        </Card.Footer>
                    </Card>
                }
                {props.error &&
                    <Alert variant="danger">
                        {props.error}
                    </Alert>
                }
            </Container>
        </div>
    )
}

function mapStateToProps(state) {
    return {
        data: state.movieReducer.detail,
        searching: state.movieReducer.searching,
        error: state.movieReducer.error,
        loading: state.movieReducer.requesting
    }
}

function mapDispatchToProps(dispatch) {
    return bindActionCreators({ fetchDetails }, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(Booking)