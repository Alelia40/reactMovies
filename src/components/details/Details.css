.detailsCard {
    width: 32rem;
    margin: 0 auto;
}