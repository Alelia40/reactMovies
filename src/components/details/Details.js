import React from 'react'

import { Container } from 'react-bootstrap';

function Details() {
    return (
        <div id="detailsWrapper">
            <Container>
                Details
            </Container>
        </div>
    )
}

export default Details
