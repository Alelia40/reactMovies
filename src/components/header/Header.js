import React from 'react'

import { Navbar, Form, FormControl, InputGroup, Button } from 'react-bootstrap';
import { FiSearch } from 'react-icons/fi';

function Header() {
    return (
        <Navbar bg="light" variant="light" expand="lg" className="justify-content-between" style={{paddingLeft: '10px', paddingRight: '10px', marginBottom: '10px'}}>
            <Navbar.Brand href="/">Movie Tickets</Navbar.Brand>
            <Form>
                <InputGroup>
                    <InputGroup.Prepend>
                        <Button variant="outline-success">
                            <FiSearch />
                        </Button>
                    </InputGroup.Prepend>
                    <FormControl placeholder="Search Movies"></FormControl>
                </InputGroup>
            </Form>
        </Navbar>
    )
}

export default Header
