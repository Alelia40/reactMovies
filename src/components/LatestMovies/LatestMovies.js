import React, { useEffect } from 'react';
import { Container, Card, Button, Row, Col } from 'react-bootstrap';
import { FiChevronRight } from 'react-icons/fi';

import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { fetchMovies } from '../../redux/actions/Index';

import './LatestMovies.css';

function LatestMovies(props) {

    useEffect(() => {
        props.fetchMovies();
    }, []);

    return (
        <Container>
            <h3>Latest Movies</h3>
            <hr></hr>
            <div className="movieDisplay">
                <Row style={{ flexWrap: 'nowrap' }}>
                    {props.loading &&
                        <Col>
                            <span>LOADING ...</span>
                        </Col>
                    }
                    {props.data &&
                        props.data.map((movie) => {
                            return (
                                <Col>
                                    <Card className="movieCard" bg="light" text="secondary" border="secondary">
                                        <Card.Img variant="top" src={movie.image} style={{ height: '400px' }} />
                                        <Card.Body>
                                            <Card.Title>{movie.title}</Card.Title>

                                            {Object.entries(movie.prices).map((seat) => {
                                                return (
                                                    <Row>
                                                        <Col>
                                                            <Button variant="outline-success" className="bookNowButton" href={`/booking/${movie.id}?seat=${seat[0]}`}>{seat[0]} seat - {seat[1]}</Button>
                                                        </Col>
                                                    </Row>
                                                )
                                            })
                                            }
                                        </Card.Body>
                                        <Card.Footer>
                                            <Card.Link className="detailsLink" href={`/details/${movie.id}`}>
                                                What's This About? <FiChevronRight />
                                            </Card.Link>
                                        </Card.Footer>
                                    </Card>
                                </Col>
                            );
                        })
                    }
                </Row>
            </div>
        </Container>
    )
}

function mapStateToProps(state) {
    return {
        data: state.movieReducer.movies,
        searching: state.movieReducer.searching,
        error: state.movieReducer.error,
        loading: state.movieReducer.requesting
    }
}

function mapDispatchToProps(dispatch) {
    return bindActionCreators({ fetchMovies }, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(LatestMovies);
