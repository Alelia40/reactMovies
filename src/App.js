import logo from './logo.svg';
import './App.css';

import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link
} from "react-router-dom";
import Header from './components/header/Header';
import Details from './components/details/Details';
import Booking from './components/booking/Booking';
import SearchPage from './components/SearchPage/SearchPage';
import LatestMovies from './components/LatestMovies/LatestMovies';
import Notfound from './components/NotFound/Notfound';

function App() {
  return (
    <Router>
      <div className="App">
        <Header></Header>
        <Switch>
          <Route exact path="/">
            <LatestMovies></LatestMovies>
          </Route>
          <Route exact path="/search">
            <SearchPage></SearchPage>
          </Route>
          <Route path="/details/:id">
            <Details></Details>
          </Route>
          <Route path="/booking/:id">
            <Booking></Booking>
          </Route>
          <Route>
            <Notfound></Notfound>
          </Route>
        </Switch>
      </div>
    </Router>
  );
}

export default App;
