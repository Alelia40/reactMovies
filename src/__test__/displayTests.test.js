import React from 'react';
import { create } from 'react-test-renderer';

import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { createStore, applyMiddleware } from 'redux';
import rootReducer from '../redux/reducers/Index';

import { BrowserRouter as Router } from 'react-router-dom';

import Booking from '../components/booking/Booking';
import Header from '../components/header/Header';
import Notfound from '../components/NotFound/Notfound';
import Details from '../components/details/Details';
import LatestMovies from '../components/LatestMovies/LatestMovies';
import SearchPage from '../components/SearchPage/SearchPage';
import App from '../App';

const store = createStore(rootReducer, applyMiddleware(thunk));

describe('testing Component snapshot', () => {
    test('header component', () => {
        let tree = create(<Header></Header>)
        expect(tree.toJSON()).toMatchSnapshot();
    })

    test('booking component', () => {
        let tree = create(
            <Provider store={store}>
                <Router>
                    <Booking></Booking>
                </Router>
            </Provider>
        )
        expect(tree.toJSON()).toMatchSnapshot();
    })

    test('details component', () => {
        let tree = create(
            <Provider store={store}>
                <Router>
                    <Details></Details>
                </Router>
            </Provider>
        )
        expect(tree.toJSON()).toMatchSnapshot();
    })

    test('latest movies component', () => {
        let tree = create(
            <Provider store={store}>
                <Router>
                    <LatestMovies></LatestMovies>
                </Router>
            </Provider>
        )
        expect(tree.toJSON()).toMatchSnapshot();
    })

    test('search page component', () => {
        let tree = create(
            <Provider store={store}>
                <Router>
                    <SearchPage></SearchPage>
                </Router>
            </Provider>
        )
        expect(tree.toJSON()).toMatchSnapshot();
    })

    test('not found component', () => {
        let tree = create(<Notfound></Notfound>)
        expect(tree.toJSON()).toMatchSnapshot();
    })

    test('app component', () => {
        let tree = create(
            <Provider store={store}>
                <Router>
                    <App></App>
                </Router>
            </Provider>
        )
        expect(tree.toJSON()).toMatchSnapshot();
    })
})