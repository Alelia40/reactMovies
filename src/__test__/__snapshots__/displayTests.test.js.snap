// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Component snapshot app component 1`] = `
<div
  className="App"
>
  <nav
    className="justify-content-between navbar navbar-expand-lg navbar-light bg-light"
    style={
      Object {
        "marginBottom": "10px",
        "paddingLeft": "10px",
        "paddingRight": "10px",
      }
    }
  >
    <a
      className="navbar-brand"
      href="/"
    >
      Movie Tickets
    </a>
    <form
      className=""
      onSubmit={[Function]}
    >
      <div
        className="input-group"
      >
        <div
          className="input-group-prepend"
        >
          <button
            className="btn btn-outline-success"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="11"
                cy="11"
                r="8"
              />
              <line
                x1="21"
                x2="16.65"
                y1="21"
                y2="16.65"
              />
            </svg>
          </button>
        </div>
        <input
          className="form-control"
          id="search"
          placeholder="Search Movies"
        />
      </div>
    </form>
  </nav>
  <div
    className="container"
  >
    <h3>
      Latest Movies
    </h3>
    <hr />
    <div
      className="movieDisplay"
    >
      <div
        className="row"
        style={
          Object {
            "flexWrap": "nowrap",
          }
        }
      >
        <div
          className="col"
        >
          <span>
            LOADING ...
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`testing Component snapshot booking component 1`] = `
<div
  id="bookingWrapper"
>
  <div
    className="container"
  >
    <span>
      LOADING ...
    </span>
  </div>
</div>
`;

exports[`testing Component snapshot details component 1`] = `
<div
  id="detailsWrapper"
>
  <div
    className="container"
  >
    <h3>
      Movie Details
    </h3>
    <hr />
    <span>
      LOADING ...
    </span>
  </div>
</div>
`;

exports[`testing Component snapshot header component 1`] = `
<nav
  className="justify-content-between navbar navbar-expand-lg navbar-light bg-light"
  style={
    Object {
      "marginBottom": "10px",
      "paddingLeft": "10px",
      "paddingRight": "10px",
    }
  }
>
  <a
    className="navbar-brand"
    href="/"
  >
    Movie Tickets
  </a>
  <form
    className=""
    onSubmit={[Function]}
  >
    <div
      className="input-group"
    >
      <div
        className="input-group-prepend"
      >
        <button
          className="btn btn-outline-success"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="11"
              cy="11"
              r="8"
            />
            <line
              x1="21"
              x2="16.65"
              y1="21"
              y2="16.65"
            />
          </svg>
        </button>
      </div>
      <input
        className="form-control"
        id="search"
        placeholder="Search Movies"
      />
    </div>
  </form>
</nav>
`;

exports[`testing Component snapshot latest movies component 1`] = `
<div
  className="container"
>
  <h3>
    Latest Movies
  </h3>
  <hr />
  <div
    className="movieDisplay"
  >
    <div
      className="row"
      style={
        Object {
          "flexWrap": "nowrap",
        }
      }
    >
      <div
        className="col"
      >
        <span>
          LOADING ...
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`testing Component snapshot not found component 1`] = `
<div
  className="container"
>
  <h3>
    404 Not Found
  </h3>
  <br />
  <a
    href="/"
  >
    Return Home
  </a>
</div>
`;

exports[`testing Component snapshot search page component 1`] = `
<div
  className="container"
>
  <h3>
    Search Results
  </h3>
  <hr />
  <div
    className="movieDisplay"
  >
    <div
      className="row"
      style={
        Object {
          "flexWrap": "nowrap",
        }
      }
    >
      <div
        className="col"
      >
        <span>
          LOADING ...
        </span>
      </div>
    </div>
  </div>
  <a
    className="btn btn-outline-success"
    href="/"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    Return To Home
  </a>
</div>
`;
